src/policy/fees.cpp

```diff
@@ -195,6 +202,45 @@ TxConfirmStats::TxConfirmStats(const std::vector<double>& defaultBuckets,
     resizeInMemoryCounters(buckets.size());
 }
 
+TxConfirmStats::TxConfirmStats(const std::vector<double>& defaultBuckets, const std::map<double, unsigned int>& defaultBucketMap, const TxConfirmStats& txstatsSource)
+    : buckets(defaultBuckets), bucketMap(defaultBucketMap)
+{
+    decay = txstatsSource.decay;
+    scale = txstatsSource.scale;
+
+    const unsigned int maxPeriods = txstatsSource.confAvg.size();
+    assert(txstatsSource.failAvg.size() == maxPeriods);
+
+    confAvg.resize(maxPeriods);
+    for (unsigned int i = 0; i < maxPeriods; i++) {
+        confAvg[i].resize(buckets.size());
+    }
+    failAvg.resize(maxPeriods);
+    for (unsigned int i = 0; i < maxPeriods; i++) {
+        failAvg[i].resize(buckets.size());
+    }
+
+    txCtAvg.resize(buckets.size());
+    avg.resize(buckets.size());
+
+    unsigned int src = 0;
+    for (unsigned int dst = 0; dst < buckets.size(); dst++) {
+        if (buckets[dst] < txstatsSource.buckets[src]) continue;
+
+        txCtAvg[dst] = txstatsSource.txCtAvg[src];
+        avg[dst] = txstatsSource.avg[src];
+        for (unsigned int i = 0; i < maxPeriods; i++) {
+            confAvg[i][dst] = txstatsSource.confAvg[i][src];
+            failAvg[i][dst] = txstatsSource.failAvg[i][src];
+        }
+
+        src++;
+        if (src >= txstatsSource.buckets.size()) break;
```

Comment:
nit: can move these conditions into the `for`, so that it manages both `src` and `dst`.